tests/integration/security/sds_tls_origination/util/util.go

```diff
@@ -0,0 +1,410 @@
+	"istio.io/istio/pkg/test/framework/components/istio"
+
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/framework/resource"
+
+	envoyAdmin "github.com/envoyproxy/go-control-plane/envoy/admin/v3"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/structpath"
+)
+
+func MustReadCert(t *testing.T, f string) string {
+	b, err := ioutil.ReadFile(path.Join(env.IstioSrc, "tests/testdata/certs/dns", f))
+	if err != nil {
+		t.Fatalf("failed to read %v: %v", f, err)
+	}
+	return string(b)
+}
+
+type TLSCredential struct {
+	PrivateKey string
+	ClientCert string
+	CaCert     string
+}
+
+const (
+	// The ID/name for the certificate chain in kubernetes tls secret.
+	tlsScrtCert = "tls.crt"
+	// The ID/name for the k8sKey in kubernetes tls secret.
+	tlsScrtKey = "tls.key"
+	// The ID/name for the CA certificate in kubernetes tls secret
+	tlsScrtCaCert = "ca.crt"
+	// The ID/name for the certificate chain in kubernetes generic secret.
+	genericScrtCert = "cert"
+	// The ID/name for the private key in kubernetes generic secret.
+	genericScrtKey = "key"
+	// The ID/name for the CA certificate in kubernetes generic secret.
+	genericScrtCaCert = "cacert"
+)
+
+// CreateKubeSecret reads credential names from credNames and key/cert from TLSCredential,
+// and creates K8s secrets for gateway.
+// nolint: interfacer
+func CreateKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string,
+	credentialType string, egressCred TLSCredential, isNotGeneric bool) {
+	t.Helper()
+	// Get namespace for gateway pod.
+	istioCfg := istio.DefaultConfigOrFail(t, ctx)
+	systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+	if len(credNames) == 0 {
+		ctx.Log("no credential names are specified, skip creating secret")
+		return
+	}
+	// Create Kubernetes secret for gateway
+	cluster := ctx.Environment().(*kube.Environment).KubeClusters[0]
+	for _, cn := range credNames {
+		secret := createSecret(credentialType, cn, systemNS.Name(), egressCred, isNotGeneric)
+		_, err := cluster.CoreV1().Secrets(systemNS.Name()).Create(context.TODO(), secret, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("Failed to create secret (error: %s)", err)
+		}
+	}
+	// Check if Kubernetes secret is ready
+	retry.UntilSuccessOrFail(t, func() error {
+		for _, cn := range credNames {
+			_, err := cluster.CoreV1().Secrets(systemNS.Name()).Get(context.TODO(), cn, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("secret %v not found: %v", cn, err)
+			}
+		}
+		return nil
+	}, retry.Timeout(time.Second*5))
+}
+
+// DeleteKubeSecret deletes a secret
+// nolint: interfacer
+func DeleteKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string) {
```

Comment:
There is a DeleteIngressKubeSecret() in tests/integration/security/sds_ingress/util/util.go, could you rename it so that we don't need another util function for deleting k8s secret?